shared/src/main/scala/coop/rchain/catscontrib/ApplicativeError_.scala

```diff
@@ -2,17 +2,11 @@ package coop.rchain.catscontrib
 
 import cats._
 
-/** Like ApplicativeError but not Applicative thus no ambigous conflicts */
-/** We should REALLY use Scalaz */
 trait ApplicativeError_[F[_], E] {
 
   implicit val instance: ApplicativeError[F, E]
 
-  def raiseError[A](e: E): F[A]                        = instance.raiseError(e)
-  def handleErrorWith[A](fa: F[A])(f: E => F[A]): F[A] = instance.handleErrorWith(fa)(f)
-  def fromEither[A](x: E Either A): F[A]               = instance.fromEither(x)
-  def attempt[A](fa: F[A]): F[Either[E, A]]            = instance.attempt(fa)
-
```

Comment:
I don't understand what is the difference between `cats.ApplicativeError` from this one. Maybe we should remove this file completely?

Reply:
I tried but not hard enough. :) I looked more carefully and found it's used in two places, `PacketHandler` and `MonadError_` which both can be fixed.

This one does not extend `Applicative` instead holds the instance as implicit field. The purpose is to prevent ambiguous implicit resolution when it's used in combination with other interfaces extending Applicative. 

`MonadError_` is used here (seems used only in tests) so it can implement `raiseError` directly.
https://github.com/rchain/rchain/blob/53b2cc19efc68aafd7e207d7b082b7af8d5c9fac/rholang/src/main/scala/coop/rchain/rholang/interpreter/matcher/package.scala#L29

And `PacketHandler` can also use `ApplicativeError` directly.
https://github.com/rchain/rchain/blob/1a90135086723203b8a590d6679abf0f91c08012/comm/src/main/scala/coop/rchain/p2p/effects/PacketHandler.scala#L19-L22

Reply:
I was just curious. We can fix this later.

Reply:
ApplicativeError_.scala is removed in https://github.com/rchain/rchain/commit/1d087b1890a91cd86ebe5522acc478c96c3e54cd.